compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantTypeChecker.java

```diff
@@ -255,6 +255,21 @@ public void visit(BLangLiteral literalExpr, AnalyzerData data) {
             return;
         }
 
+        if (literalType.tag == TypeTags.BYTE_ARRAY) {
+            BLangListConstructorExpr listConstructorExpr = rewriteByteArrayLiteral(literalExpr);
+            BType expType = Types.getReferredType(data.expType);
+            if (expType.tag == TypeTags.ARRAY && ((BArrayType) expType).state == BArrayState.INFERRED) {
+                ((BArrayType) expType).size = listConstructorExpr.exprs.size();
+                ((BArrayType) expType).state = BArrayState.CLOSED;
+            }
+            if (types.typeIncompatible(literalExpr.pos, listConstructorExpr.getBType(), data.expType)) {
+                data.resultType = symTable.semanticError;
+                return;
+            }
+            data.resultType = checkListConstructorCompatibility(literalType, listConstructorExpr, data);
```

Comment:
As far as I understood, first we create a `BLangListConstructorExpr ` and use that return tuple type here. But can't we directly create tuple type at #L259. It seems unnecessary step to me. Is there any use of the created ListConstructorExpr other than passing it to this method?

Reply:
+1, we should be able return the tuple type and just check type compatibility. Will update.